pilot/adapter/config/file/monitor_test.go

```diff
@@ -0,0 +1,259 @@
+			// Get the configuration that we expect to be added.
+			expectedCfg, err := parseConfig("testdata/" + fileName)
+			if err != nil {
+				t.Fatalf("Unable to parse config: %s", fileName)
+				return
+			}
+
+			// Add the configuration.
+			if err := copyTestFile(fileName, e.fsroot); err != nil {
+				t.Fatalf("Failed adding config file: %s", fileName)
+				return
+			}
+
+			// Wait for the add to occur.
+			event := e.watch()
+
+			// Verify that the configuration was added properly.
+			if event.event != model.EventAdd {
+				t.Fatalf("Expected add event: %s", event)
+				return
+			}
+
+			// Need to clear the resource version before comparing.
+			if !reflect.DeepEqual(*expectedCfg, event.config) {
+				t.Fatalf("Config does not match.\nExpected:\n%s\nReceived:\n%s", *expectedCfg, event.config)
+			}
+		})
+	}
+}
+
+func TestDelete(t *testing.T) {
+	// Copy all of the files over to the temp dir before initializing the environment.
+	tempDir := createTempDir()
+	for _, fileName := range files {
+		if err := copyTestFile(fileName, tempDir); err != nil {
+			t.Fatalf("Failed adding config file: %s", fileName)
+			return
+		}
+	}
+
+	e := env{}
+	e.setup(tempDir)
+	defer e.teardown()
+
+	for _, fileName := range files {
+		t.Run(fileName, func(t *testing.T) {
+			// Get the configuration that we expect to be deleted
+			expectedCfg, err := parseConfig("testdata/" + fileName)
+			if err != nil {
+				t.Fatalf("Unable to parse config: %s", fileName)
+				return
+			}
+
+			err = os.Remove(e.fsroot + "/" + fileName)
+			if err != nil {
+				t.Fatalf("Unable to remove file: %s", fileName)
+			}
+
+			event := e.watchFor(model.EventDelete)
+
+			// Need to clear the resource version before comparing.
+			event.config.ResourceVersion = ""
+			if !reflect.DeepEqual(*expectedCfg, event.config) {
+				t.Fatalf("Config does not match.\nExpected:\n%s\nReceived:\n%s", *expectedCfg, event.config)
+			}
+		})
+	}
+}
+
+func TestUpdate(t *testing.T) {
+	// Copy all of the files over to the temp dir before initializing the environment.
+	tempDir := createTempDir()
+	for _, fileName := range files {
+		if err := copyTestFile(fileName, tempDir); err != nil {
+			t.Fatalf("Failed adding config file: %s", fileName)
+			return
+		}
+	}
+
+	e := env{}
+	e.setup(tempDir)
+	defer e.teardown()
+
+	// Overwrite the original file with the updated.
+	updateFilePath := "testdata/cb-policy.yaml.update"
+	copyFile(updateFilePath, e.fsroot+"/"+"cb-policy.yaml")
+
+	// Get the configuration that we expect to be deleted
+	expectedCfg, err := parseConfig(updateFilePath)
+	if err != nil {
+		t.Fatalf("Unable to parse config: %s", updateFilePath)
+		return
+	}
+
+	event := e.watchFor(model.EventUpdate)
+
+	// Need to clear the resource version before comparing.
+	event.config.ResourceVersion = ""
+	if !reflect.DeepEqual(*expectedCfg, event.config) {
+		t.Fatalf("Config does not match.\nExpected:\n%s\nReceived:\n%s", *expectedCfg, event.config)
```

Comment:
In golang we tend to use the style `Got...Want...`. So instead of this failure message, do something like:

`t.Fatalf("event.config = %v, wanted %q", event.config, expectedCfg)`